app/assets/javascripts/student_profile/TransitionNotes.js

```diff
@@ -67,10 +67,22 @@ class TransitionNotes extends React.Component {
     this.onChangeRestrictedNote = this.onChangeRestrictedNote.bind(this);
   }
 
+  // Note on componentWillReceiveProps:
   // This method will be deprecated in React 16.3. But we're still using
-  // React 15.4 and have some migration work to finish up til we can migrate to
-  // React 16. Just noting that this will eventually have to be refactored
-  // eventually.
+  // React 15.4 in Insights, and have some legacy-cleanup work to finish
+  // til we can migrate to 16. Noting that this func will have to be refactored.
+
+  // In my local environment, when I type into the Transition Note textarea,
+  // the server responds with "saved!" so fast that the status text is barely
+  // readable when it's supposed to say "Autosaving..."
+
+  // This code adds an artificial delay so that the text that says "Autosaving..."
+  // can hang for around 2 seconds to let the user read it.
+
+  // It occurred to me that the server might respond more slowly on the Heroku app,
+  // which would make this code unnecessary. So I'm going to try commenting it out,
+  // deploying up to the Heroku demo site, and seeing what the user experience
+  // is like.
   componentWillReceiveProps(newProps) {
```

Comment:
In general `componentDidUpdate` is the better hook for executing some kind of side-effect on a state/prop change.  You can read more here if you're curious: https://reactjs.org/docs/react-component.html#unsafe_componentwillreceiveprops, but should be able to just swap method and args and keep your same code.

The other thing that allows is to remove the manual autosave calls, and just save every time the state changes in ways you care about.  There's an example [here](https://github.com/studentinsights/studentinsights/blob/master/app/assets/javascripts/class_lists/ClassListCreatorPage.js#L116) if that's helpful - the other trick is to `_.throttle` or `_.debounce` calls to the [actual save method](https://github.com/studentinsights/studentinsights/blob/master/app/assets/javascripts/class_lists/ClassListCreatorPage.js#L84).

Reply:
When I just swap in `componentDidUpdate` as the function name, it works as expected. When I try to update the args too and reflect the fact that one passes in `newProps` and the other, `prevProps`, it stops working as expected. 🤔 🤔 🤔

Reply:
@alexsoble sounds good!  I'm happy to look at this with you sometime next week, in person even :)